src/common/options.cc

```diff
@@ -6899,6 +6899,10 @@ std::vector<Option> get_mds_options() {
     .set_default(5)
     .set_description("minimum interval between scatter lock updates"),
 
+    Option("mds_client_msg_throttle", Option::TYPE_SIZE, Option::LEVEL_ADVANCED)
+    .set_default(1<<16)
```

Comment:
Am I missing something or does this allocate 64KiB for all client messages?

As Sage noted, it is per-policy, not per-client. And I'm pretty sure this unit is in bytes...
Even ignoring that, I'd probably make it much larger since it really just needs to be smaller than the total dispatch throttle, and we do have the inline data option.

Reply:
> Am I missing something or does this allocate 64KiB for all client messages?

This would be a throttle of 64KiB for each client session. (At least, that's the intent. Maybe not based on sage's comment.)

Reply:
Ah, I misread it as 64 MB... which is probably what I'd suggest setting this to (using the 64_MB syntax instead of << to avoid confusion).  We don't have per-connection throttles.. though perhaps we should!

Reply:
Can we get some napkin math on how many messages that actually is?
I have it in my head that an MDS in-memory inode is about 1.5KiB, which means that if a Client is flushing back dirty caps then 64KiB per-client would probably be waaaay too small. 64MiB for all clients is obviously more useful, but at 1k clients that's the same total number...